rust/worker/src/index/hnsw_provider.rs

```diff
@@ -328,8 +347,8 @@ impl HnswIndexProvider {
         // Scope to drop the cache lock before we await to write to s3
         // TODO: since we commit(), we don't need to save the index here
         {
-            let cache = self.cache.read();
-            let index = match cache.get(id) {
+            // let cache = self.cache.read();
+            let index = match self.cache.get(id) {
```

Comment:
i think this has a similar problem where if the cache evicts a flushed index ?

Reply:
Yes. We need to do something about it.